rocketpy/mathutils/function.py

```diff
@@ -2612,6 +2622,77 @@ def compose(self, func, extrapolate=False):
                 extrapolation=self.__extrapolation__,
             )
 
+    def _check_user_input(
+        self,
+        source,
+        inputs,
+        outputs,
+        interpolation,
+        extrapolation,
+    ):
+        # check source for data type
+
+        # if list or ndarray, check for dimensions, interpolation and extrapolation
+        if isinstance(source, (list, np.ndarray)):
+            # this will also trigger an error if the source is not a list of numbers or if the array is not homogeneous
+            source = np.array(source, dtype=np.float64)
+
+            # check dimensions
+            source_dim = source.shape[1]
+
+            # check output type and dimensions
+            if isinstance(outputs, str):
```

Comment:
I think these two lines should be at a higher level of this method. I suggest moving this to line 2634.

Also, you can do the same thing for the inputs. You're gonna see that this will fix some tests ;)

Reply:
You cannot make `source_dim = source.shape[1]` if the input is a scalar, which currently is a supported inputs from function class